platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/userlocation/MockLocationEngine.java

```diff
@@ -1,38 +1,52 @@
 package com.mapbox.mapboxsdk.testapp.activity.userlocation;
 
-
+import android.animation.AnimatorListenerAdapter;
+import android.animation.TypeEvaluator;
+import android.animation.ValueAnimator;
 import android.location.Location;
-import android.os.Handler;
 
 import com.mapbox.services.android.telemetry.location.LocationEngine;
 import com.mapbox.services.android.telemetry.location.LocationEngineListener;
 
 /**
- * Sample LocationEngine that provides mocked locations simulating GPS updates
+ * Sample LocationEngine that provides mocked LOCATIONS simulating GPS updates
  */
 public class MockLocationEngine extends LocationEngine {
+  private static MockLocationEngine INSTANCE;
+
+  private final LocationAnimator locationAnimator;
+  private boolean running;
+
+  MockLocationEngine(Location start, Location end) {
+    locationAnimator = new LocationAnimator(start, end, new ValueAnimator.AnimatorUpdateListener() {
+      @Override
+      public void onAnimationUpdate(ValueAnimator animation) {
+        for (LocationEngineListener listener : locationListeners) {
+          listener.onLocationChanged((Location) animation.getAnimatedValue());
+        }
+      }
+    });
+  }
 
-  // Mocked data
-  private static final int UPDATE_INTERVAL_MS = 1000;
-  private static final double[][] locations = new double[][] {
-    new double[] {39.489309, -0.360415},
-    new double[] {39.492469, -0.358777},
-    new double[] {40.393285, -3.707260},
-    new double[] {40.394374, -3.707767},
-    new double[] {40.398012, -3.715943},
-    new double[] {40.416913, -3.703861}};
-
-  private Handler handler;
-  int currentIndex;
-
-  public MockLocationEngine() {
-    super();
+  public static synchronized MockLocationEngine getInstance() {
+    if (INSTANCE == null) {
+      INSTANCE = new MockLocationEngine(
+        MockLocationEngine.createLocation(40.416913, -3.703861),
```

Comment:
We aren't using less, but actually using more. This MockLocationEngine is being powered by a ObjectAnimator that animates the location position over time. It produces 604 location updates to be precise.